src/lib/client.ts

```diff
@@ -63,16 +63,19 @@ export class Client {
     this.unsubscribe = await this.api.rpc.chain.subscribeNewHeads(
       async ({ hash, number }) => {
         const blockNum = number.toNumber()
-        const signedBlock = await this.api.rpc.chain.getBlock(hash)
-        const preRuntimeLog = signedBlock.block.header.digest.logs.find((digestItem) => digestItem.isPreRuntime)?.asPreRuntime[1];
+        const header = await this.api.rpc.chain.getHeader(hash);
+
+        // TODO: handle vote rewards: check VoteReward events and aggregate
```

Comment:
Voting rewards are in the transactions and you shouldn't need to read blocks for that, if you can just go through events it'll tell you everything you need to know.